src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2590,16 +2629,107 @@ public override object PayloadValue(int index)
             }
         }
 
-        private event Action<GenAwareBeginTraceData> m_target;
+        private event Action<GenAwareTemplateTraceData> m_target;
         #endregion
     }
-    public sealed class GenAwareEndTraceData : TraceEvent
+
+    public sealed class GCLOHCompactInfo
     {
-        public int Count { get { return GetInt32At(0); } }
-        public int ClrInstanceID { get { return GetInt16At(4); } }
+        /// <summary>
+        /// The time spent on planing the LOH for compaction
+        /// </summary>
+        public int TimePlan
+        {
+            get
+            {
+                return m_timePlan;
+
+            }
+        }
+
+        /// <summary>
+        /// The time spent on copying objects during LOH compaction
+        /// </summary>
+        public int TimeCompact
+        {
+            get
+            {
+                return m_timeCompact;
+
+            }
+        }
+
+        /// <summary>
+        /// The time spent on relocating pointers during LOH compaction
+        /// </summary>
+        public int TimeRelocate
+        {
+            get
+            {
+                return m_timeRelocate;
+            }
+        }
+
+        /// <summary>
+        /// The total number of pointers found in the LOH during LOH compaction
+        /// </summary>
+        public long TotalRefs
+        {
+            get
+            {
+                return m_totalRefs;
+            }
+        }
+
+        /// <summary>
+        /// The total number of pointers found pointing to null in the LOH during LOH compaction
+        /// </summary>
+        public long ZeroRefs
+        {
+            get
+            {
+                return m_zeroRefs;
+            }
+        }
+
+        #region private
+        internal GCLOHCompactInfo(int timePlan, int timeCompact, int timeRelocate, long totalRefs, long zeroRefs)
+        {
+            m_timePlan = timePlan;
+            m_timeCompact = timeCompact;
+            m_timeRelocate = timeRelocate;
+            m_totalRefs = totalRefs;
+            m_zeroRefs = zeroRefs;
+        }
+
+        private int m_timePlan;
+        private int m_timeCompact;
+        private int m_timeRelocate;
+        private long m_totalRefs;
+        private long m_zeroRefs;
+        #endregion
+    }
+
+    public class GCLOHCompactTraceData : TraceEvent
```

Comment:
shouldn't this be a sealed class like all the other *TraceData classes?

Reply:
Yes, please do this.